src/harness/testcases/WINNF_FT_S_FDB_testcase.py

```diff
@@ -0,0 +1,1240 @@
+    # Set the grant frequency to overlap with the FSS 'KA413' which is 3625-4200 MHz.
+    grant_g_a['operationParam']['operationFrequencyRange'] = {
+        'lowFrequency': 3650000000,
+        'highFrequency': 3660000000
+    }
+
+    grant_g_b = json.load(
+        open(os.path.join('testcases', 'testdata', 'grant_0.json')))
+    # Set the grant frequency to overlap with the FSS 'E000306' which is 3625-3700 MHz.
+    grant_g_b['operationParam']['operationFrequencyRange'] = {
+        'lowFrequency': 3670000000,
+        'highFrequency': 3680000000
+    }
+
+    # Update the location 'X' of CBSD devices to be near FSS sites
+    # 'KA413' at Albright,WV
+    device_a['installationParam']['latitude'] = 39.57327
+    device_a['installationParam']['longitude'] = -79.61903
+
+    # 'E000306' at Andover,ME
+    device_b['installationParam']['latitude'] = 44.63367
+    device_b['installationParam']['longitude'] = -70.69758
+
+    # Creating conditionals for Cat B devices
+    self.assertEqual(device_b['cbsdCategory'], 'B')
+    conditionals_b = {
+        'cbsdCategory': device_b['cbsdCategory'],
+        'fccId': device_b['fccId'],
+        'cbsdSerialNumber': device_b['cbsdSerialNumber'],
+        'airInterface': device_b['airInterface'],
+        'installationParam': device_b['installationParam'],
+        'measCapability': device_b['measCapability']
+    }
+    del device_b['cbsdCategory']
+    del device_b['airInterface']
+    del device_b['installationParam']
+    del device_b['measCapability']
+
+    conditionals = [conditionals_b]
+
+    # Create the actual config.
+    config = {
+        'registrationRequests': [device_a, device_b],
+        'grantRequests': [grant_g_a, grant_g_b],
+        'conditionalRegistrationData': conditionals
+        # TODO
+        # Need to add data base configurations
+    }
+    writeConfig(filename, config)
+
+  @configurable_testcase(generate_FDB_8_default_config)
+  def test_WINNF_FT_S_FDB_8(self, config_filename):
+    """FSS Database Update: Scheduled Database Update."""
+
+    # Load the configuration file
+    config = loadConfig(config_filename)
+
+    # Step 1: Trigger daily activities
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
+
+    # Step 2: Registration of CBSDs and requesting for grants.
+    # Forming of 'G' grants for all the configured grants
+    grant_request_g = config['grantRequests']
+
+    # Register device(s) 'C' and request grant 'G' with SAS UUT.
+    cbsd_ids, grant_ids = self.assertRegisteredAndGranted(config['registrationRequests'],
+                                    grant_request_g,
+                                    config['conditionalRegistrationData'])
+
+    # TODO
+    # Step 3: Create FSS database which includes atleast one FSS site near location 'X'.
+
+    # TODO
+    # Step 4: Inject the FSS database URL into the UUT
+
+    # Step 5: Trigger daily activities
+    self.TriggerDailyActivitiesImmediatelyAndWaitUntilComplete()
```

Comment:
You need to wait until a pre-specified time, not trigger CPAS.

Reply:
FDB.8     Added 'cpasInitiateTimer': 100, in config data structure     and     sleep(config['cpasInitiateTimer']) in test case definition step 5.

Reply:
I think you misunderstand: this test case it to check that the SAS executes CPAS at the correct time *without* being triggered, so you *cannot* trigger it to run CPAS in this test case.